beginner_source/ax_multiobjective_nas_tutorial.py

```diff
@@ -0,0 +1,524 @@
+- `BoTorch <https://github.com/pytorch/botorch>`__ (the Bayesian Optimization library powering Ax's algorithms)
+"""
+
+
+######################################################################
+# Defining the TorchX App
+# -----------------------
+#
+# Our goal is to optimize the PyTorch Lightning training job defined
+# in `mnist_train_nas.py <https://github.com/pytorch/tutorials/tree/master/beginner_source/mnist_train_nas.py>`__.
+# To do this using TorchX, we write a helper function that takes in
+# the values of the architcture and hyperparameters of the training
+# job and creates a TorchX AppDef (see the
+# `TorchX docs <https://pytorch.org/torchx/latest/basics.html>`__
+# for more detail) with the appropriate settings.
+#
+
+
+from pathlib import Path
+
+from torchx import specs
+from torchx.version import TORCHX_IMAGE
+
+
+# load in the training code
+with Path.cwd().joinpath("mnist_train_nas.py").open("rt") as f:
+    train_source = f.read()
+
+
+def trainer(
+    log_path: str,
+    hidden_size_1: int,
+    hidden_size_2: int,
+    learning_rate: float,
+    epochs: int,
+    dropout: float,
+    batch_size: int,
+    trial_idx: int = -1,
+) -> specs.AppDef:
+
+    # define the log path so we can pass it to the TorchX AppDef
+    if trial_idx >= 0:
+        log_path = Path(log_path).joinpath(str(trial_idx)).absolute().as_posix()
+
+    return specs.AppDef(
+        name="trainer",
+        roles=[
+            specs.Role(
+                name="trainer",
+                image=TORCHX_IMAGE,
+                entrypoint="python",
+                args=[
+                    "-c",
+                    train_source,
+                    "--log_path",
+                    log_path,
+                    "--hidden_size_1",
+                    str(hidden_size_1),
+                    "--hidden_size_2",
+                    str(hidden_size_2),
+                    "--learning_rate",
+                    str(learning_rate),
+                    "--epochs",
+                    str(epochs),
+                    "--dropout",
+                    str(dropout),
+                    "--batch_size",
+                    str(batch_size),
+                ],
+            )
+        ],
+    )
+
+
+######################################################################
+# Setting up the Runner
+# ---------------------
+#
+# Ax’s `Runner <https://ax.dev/api/core.html#module-ax.core.runner>`__
+# abstraction allows writing interfaces to various backends.
+# Ax already comes with Runner for TorchX, and so we just need to
+# configure it. For the purpose of this tutorial we run jobs locally
+# in a fully asynchronous fashion.
+#
+# In order to launch them on a cluster, you can instead specify a
+# different TorchX scheduler and adjust the configuration appropriately
+# (e.g. if you have a Kubernetes cluster, you just need to change the
+# scheduler from local_cwd to kubernetes).
```

Comment:
```suggestion
# scheduler `from local_cwd` to `kubernetes`.
```

Reply:
I think you need double ` for this to render as code
```suggestion
# scheduler from ``local_cwd`` to ``kubernetes``).
```